homeassistant/components/overkiz/config_flow.py

```diff
@@ -132,14 +259,91 @@ async def async_step_user(
+                user_input = await self.async_validate_input(user_input)
+            except TooManyRequestsException:
+                errors["base"] = "too_many_requests"
+            except BadCredentialsException:
+                errors["base"] = "invalid_auth"
+            except (TimeoutError, ClientError, ClientConnectorError):
+                errors["base"] = "cannot_connect"
+            except MaintenanceException:
+                errors["base"] = "server_in_maintenance"
+            except TooManyAttemptsBannedException:
+                errors["base"] = "too_many_attempts"
+            except NotSuchTokenException:
+                errors["base"] = "not_such_token"
+            except DeveloperModeDisabled:
+                errors["base"] = "developer_mode_disabled"
+            except UnknownUserException:
+                # Somfy Protect accounts are not supported since they don't use
+                # the Overkiz API server. Login will return unknown user.
+                description_placeholders["unsupported_device"] = "Somfy Protect"
+                errors["base"] = "unsupported_hardware"
+            except Exception as exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+                LOGGER.exception(exception)
+            else:
+                if self._config_entry:
+                    if self._config_entry.unique_id != self.unique_id:
+                        return self.async_abort(reason="reauth_wrong_account")
+
+                    # Update existing entry during reauth
+                    self.hass.config_entries.async_update_entry(
+                        self._config_entry,
+                        data={
+                            **self._config_entry.data,
+                            **user_input,
+                        },
+                    )
+
+                    self.hass.async_create_task(
+                        self.hass.config_entries.async_reload(
+                            self._config_entry.entry_id
+                        )
+                    )
+
+                    return self.async_abort(reason="reauth_successful")
+
+                # Create new entry
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=user_input[CONF_HOST], data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="local",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=self._host): str,
+                    vol.Required(CONF_USERNAME, default=self._user): str,
                     vol.Required(CONF_PASSWORD): str,
-                    vol.Required(CONF_HUB, default=self._default_hub): vol.In(
-                        {key: hub.name for key, hub in SUPPORTED_SERVERS.items()}
-                    ),
                 }
             ),
             description_placeholders=description_placeholders,
```

Comment:
Not sure if this was already an issue with the integration or if I messed up applying this series (using `curl https://github.com/home-assistant/core/pull/71644.patch | patch -p1` within the `home-assistant` docker container on HASS OS), but I don't seem to get field names/placeholders on either the local or cloud API configuration screen:

![image](https://github.com/home-assistant/core/assets/2325264/ebc2909b-74e6-48bf-a5f8-eff0664dbcd9)

(Only the default gateway placeholder as configured below is shown)

Reply:
This should work in Home Assistant Core, thus I think this is due to your method of applying the patch.

Reply:
Then what is the canonical way for _users_ to manually install PRs with changes to _existing integrations_ to their home assistant instance for testing (and of course active use of the added features/fixes)? I only found https://community.home-assistant.io/t/testing-pull-requests-via-custom-components/251587 and copied the `overkiz` folder from a local git checkout on your PR to `custom_components/overkiz/`, added a `"version"` field to `manifest.json` to get it to load but the placeholders in these fields are still missing. Everything else is working as expected, and this method appears to be more convenient as I no longer have to patch the container whenever there are HASS core updates.

Perhaps the docker image is built with some sort of cache or other output over `strings.json`, such that your changes to it don't get reflected on my instance?

Is it worth trying out https://github.com/alandtse/pr_custom_component (as you suggested in https://community.home-assistant.io/t/overkiz-api-and-somfy-api/61448/1824) to see if that resolves this issue?

Reply:
Just check out this specific branch and run Home Assistant. This is the only way for developers to run a specific version. I cannot support other ways, since that is not related to this PR.

Reply:
Note that I'm in no way asking you to support "other ways", merely inquiring what _the way_ is.

Indeed, when running this PR in a `devcontainer` the placeholders are available. It was immediately obvious from seeing `Generating translations for xxx` in the build logs that this must have been missing; copying the generated `homeassistant/components/overkiz/translations/` over to the `custom_components` folders solves it :tada: (still surprised that the rest of the text was working normally).

So, for next time I'll regenerate this when pulling in this PR to my instance. Thanks for pointing me in the right direction. TIL.